piptools/resolver.py

```diff
@@ -100,34 +109,36 @@ def combine_install_requirements(repository, ireqs):
 class Resolver:
     def __init__(
         self,
-        constraints,
-        repository,
-        cache,
-        prereleases=False,
-        clear_caches=False,
-        allow_unsafe=False,
-    ):
+        constraints: List[InstallRequirement],
```

Comment:
Can relax the `List` as the line coerces to a `Set` anyway.

```suggestion
        constraints: Iterable[InstallRequirement],
```

Reply:
Indeed. Thanks, addressed in 386c8e07c054176758ed65a937c9fbc9dad0c1d2.